gcloud/datastore/batch.py

```diff
@@ -159,7 +163,11 @@ def commit(self):
         however it can be called explicitly if you don't want to use a
         context manager.
         """
-        self.connection.commit(self._dataset_id, self.mutation)
+        response = self.connection.commit(self._dataset_id, self.mutation)
+        for new_key_pb, entity in zip(response.insert_auto_id_key,
+                                      self._auto_id_entities):
```

Comment:
@pcostell Are we gauranteed that a [`MutationResult`][1] will have the correct keys in the same order sent in the original `Mutation`? (i.e. Can the order change? Can there be partial success?)

[1]: https://github.com/GoogleCloudPlatform/google-cloud-datastore/blob/21e029b2657ebb15fef5429c90c4315326855fd1/proto/datastore_v1.proto#L409